server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -626,6 +626,10 @@ protected Node(
             additionalSettings.add(NODE_MASTER_SETTING);
             additionalSettings.add(NODE_REMOTE_CLUSTER_CLIENT);
             additionalSettings.addAll(pluginsService.getPluginSettings());
+            for (Setting<Boolean> ff : pluginsService.getPluginFeatureFlags()) {
+                FeatureFlags.registerFeatureFlag(ff);
```

Comment:
Could you get rid of this static mutable API and instead just pass a collection of all the plugin feature flags to `FeatureFlags.initialize`?